include/fastrtps/rtps/participant/RTPSParticipant.h

```diff
@@ -141,6 +141,8 @@ class RTPS_DllAPI RTPSParticipant
 
     bool get_remote_reader_info(const GUID_t& readerGuid, ReaderProxyData& returnedInfo);
 
+    ResourceEvent& get_resource_event() const;
```

Comment:
Why this is needed at Public API level?

Reply:
This one I believe is needed: 
- to initialize the `TimedCallback` I need to get the resource event, which is owned by `RTPSParticipantImpl`
- but `TimedCallback` lives in `PublisherImpl`, which only has access to `ParticipantImpl`

So I needed to get the resource like this:

```
ParticipantImpl::get_resource_event()
    -> RTPSParticipant::get_resource_event()
        -> RTPSParticipantImpl::get_resource_event()
            -> return resource_event;
```

Is there an alternative? This is the same I did in the Deadline QoS